File: app/src/main/java/com/notes/ayatkhraisat/mvppratctice/di/ApplicationComponent.java
package com.notes.ayatkhraisat.mvppratctice.di;

import android.app.Application;
import com.notes.ayatkhraisat.mvppratctice.di.presenter.PresenterModule;
import com.notes.ayatkhraisat.mvppratctice.di.presenter.RepoModule;
import com.notes.ayatkhraisat.mvppratctice.top_rated.TopRatedMoviesActivity;
import dagger.Component;
import dagger.android.DaggerApplication;

import javax.inject.Singleton;

@Singleton
@Component(modules = {RepoModule.class, PresenterModule.class})
public interface ApplicationComponent {

    void inject(TopRatedMoviesActivity topRatedMoviesActivity);





}

File: app/src/main/java/com/ayatkhraisat/mvp_example/di/components/ApplicationComponent.kt
package com.ayatkhraisat.mvp_example.di.components

import android.content.Context
import com.ayatkhraisat.mvp_example.App
import com.ayatkhraisat.mvp_example.di.modules.ApplicationModule
import com.ayatkhraisat.mvp_example.di.modules.NetworkModule
import com.ayatkhraisat.mvp_example.di.qualifires.ApplicationContext
import com.ayatkhraisat.mvp_example.network.MoviesService
import dagger.Component
import javax.inject.Singleton

/**
 * Created by Ayat Khriasat on 14,December,2018 at 7:33 PM
 * Email: ayatzkhraisat@gmail.com
 * Project: MvpExample
 */

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, NetworkModule::class))
interface ApplicationComponent {

    fun inject(application: App)

    @ApplicationContext
    fun getApplication(): Context

    fun getNetworkService(): MoviesService

}
